dev/docker/README.md

```diff
@@ -0,0 +1,34 @@
+### Developing with Docker
+
+The intention of the Docker configurations in this directory is to enable anyone to develop and test NuCypher on all major operating systems with minimal prerequisites and installation hassle.
```

Comment:
@vepkenez - What do you think about moving this work into either `deploy/docker` or `examples/something`?

Reply:
Yeah... I am open to anything and was waiting for some suggestions like this.  The thing is, it is totally distinct from *deploy* right now.  Like... this is for development.  The /docker inside deploy is different and will be developed differently.

Reply:
This part is still unclear to me - it seems the only distinction is only slight, right? (which network ports and domains are being used, for example).  When developing dApps, I mostly imagine the running development fleet as the background to the tests, nearly indistinguishable from production Ursulas.

Reply:
The way it is installed is totally different.  This is all about `-e`.  The regular docker deploy actually pulls our docker image just does some basic setup.